ratelimiting/ratelimiting_user.c

```diff
@@ -227,18 +230,75 @@ static void update_ports(char *ports)
     uint16_t port = 0;
     uint8_t pval = 1;
     tmp = strdup(ports);
-    while((ptr = strsep(&tmp, delim)) != NULL)
-    {
+    ptr = strtok(tmp, delim);
+
+    while (ptr != NULL) {
         ptr = trim_space(ptr);
         port = (uint16_t)(strtoi(ptr));
+        printf("Adding port %d to map \n",port);
         bpf_map_update_elem(map_fd[4], &port, &pval, 0);
+        ptr = strtok(NULL, delim);
     }
+
+    /*while((ptr = strsep(&tmp, delim)) != NULL)
+    {
+        ptr = trim_space(ptr);
+        port = (uint16_t)(strtoi(ptr));
+        bpf_map_update_elem(map_fd[4], &port, &pval, 0);
+    }*/
+
     free(tmp);
 }
 
+static int // 0 on success and sets prog_fd and map_fd, non-zero on failure and sets bpf_log_buf.
+load_ratelimiting_bpf_file(const char* filename)
```

Comment:
load_bpf_file function under bpf_load.c wasn't loading all the map_fds so created a separate function here

Reply:
bpf_load.c is loading only two fixed maps. Since bpf_load.c is being shared with other sample ebpf programs, I didn't modify the function load_bpf_file, instead created a new one.